src/lib/scheduler/worker.cpp

```diff
@@ -68,10 +77,9 @@ void Worker::_work() {
       }
     }
 
-    // Sleep if there is no ready task in our queue and work stealing was not successful.
+    // Return if there is no ready task in our queue and work stealing was not successful.
     if (!work_stealing_successful) {
-      std::this_thread::sleep_for(std::chrono::milliseconds(10));
-      return;
+      return true;
```

Comment:
I would maybe move the whole wait-for-condition-variable-thing to here, where previously the sleeping happened. That way you don't actually have to check whether the queues are empty or not, because that is implicit: If we reached this point, then a) there was no task in my queue, and b) we could not steal a task. Therefore we then go into wait mode until notified again.